c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -842,9 +842,16 @@ mfxStatus MfxC2DecoderComponent::InitDecoder(std::shared_ptr<C2BlockPool> c2_all
             mfx_res = m_mfxDecoder->QueryIOSurf(&m_mfxVideoParams, &decRequest);
             if (MFX_ERR_NONE == mfx_res) {
                if (m_uOutputDelay < decRequest.NumFrameSuggested) {
-                   MFX_DEBUG_TRACE_MSG("More buffer needed for decoder output!");
-                   ALOGE("More buffer needed for decoder output! Actual: %d. Expected: %d", m_uOutputDelay, decRequest.NumFrameSuggested);
-                   mfx_res = MFX_ERR_MORE_SURFACE;
+                    MFX_DEBUG_TRACE_MSG("More buffer needed for decoder output!");
+                    ALOGW("More buffer needed for decoder output! Actual: %d. Expected: %d, update to %d", 
+                        m_uOutputDelay, decRequest.NumFrameSuggested, decRequest.NumFrameSuggested);\
+
+                    m_uOutputDelay = decRequest.NumFrameSuggested;
+                    m_updatingC2Configures.push_back(C2Param::Copy(C2PortDelayTuning::output((C2Uint32Value)m_uOutputDelay)));
+
+                    m_paramStorage.UpdateValue(C2PortDelayTuning::output::PARAM_TYPE, 
```

Comment:
Please do not change the logic here. Although outputDelay can be changed in framework by calling configUpdate. But buffer allocation in codec2.0 https://github.com/projectceladon/MediaSDK_C2/blob/master/c2_utils/src/mfx_va_frame_pool_allocator.cpp#L33 is one time allocation unless we re-init decoder, which usually called when format change happened. 
Just changing this outputDelay value may result in buffers mismatch between framework and mediasdk. That's why we give the maximum buffer count when component created, and outputDelay is set the maximum buffer count.

If you happen run into a case that the outputDelay is less than the suggested buffer count, just increase the outputDelay value when component created.

Reply:
I think change of this value is before allocating buffer.
The change is on line 849, and set is on line 919 in same function.
https://github.com/projectceladon/MediaSDK_C2/blob/64603f78f38fec02039414dcc5bb910993b15fb5/c2_components/src/mfx_c2_decoder_component.cpp#L911

Reply:
The outputDelay is used to set the maximum dequeued buffer in aosp. https://github.com/projectceladon/MediaSDK_C2/blob/master/c2_utils/src/mfx_va_frame_pool_allocator.cpp#L94 is not running in the same thread as https://cactus-source.iind.intel.com/source/xref/CELADON-MASTER-R2/frameworks/av/media/codec2/sfplugin/CCodecBufferChannel.cpp#1639, which means change of outputDelay in mediasdk c2 doesn't work timely. We may still allocate the previous buffer count in https://github.com/projectceladon/MediaSDK_C2/blob/master/c2_utils/src/mfx_va_frame_pool_allocator.cpp#L94, which may cause buffer mismatch between aosp and mediasdk c2.

Reply:
I don't think aosp and mediasdk must be consistent in all times. I also said before, if Ignore this mismatch, it can still works fine usually. Even if an error occurs, why not return the error where the error occurred?

Reply:
I suggest increasing the outputDelay value while creating the component instead of updating this value while decoding thread starts.